### PR TITLE
Implement EffectManager::SpawnParticles

### PR DESCRIPTION
**Pull request by Renzo904**

### CODE REVIEW

File: src/EffectManager.cpp [hunk 1]
```diff
@@ -135,3 +158,58 @@ ChainCallbackResult EffectManager::OnDraw(EffectManager *mgr)
...
+    for (idx = 0; idx < ARRAY_SIZE_SIGNED(this->effects); idx++)
+    {
+        this->nextIndex++;
+        if (this->nextIndex >= ARRAY_SIZE_SIGNED(this->effects))
+        {
+            this->nextIndex = 0;
+        }
+        if (effect->inUseFlag)
+        {
+            if (!this->nextIndex)
```
Comment (roblabla) [inline src/EffectManager.cpp:178]:
```suggestion
            if (this->nextIndex == 0) {
                effect = &this->effects[0]
            }
```

I think this is a bit easier to understand what's going on.

File: src/EffectManager.cpp [hunk 2]
```diff
@@ -135,3 +158,58 @@ ChainCallbackResult EffectManager::OnDraw(EffectManager *mgr)
...
+            {
+                effect = this->effects;
+            }
```
Comment (roblabla) [inline src/EffectManager.cpp:181]:
```suggestion
            if (this->nextIndex == 0)
            {
                effect = &this->effects[0]
            }
```

I think this is a bit easier to understand what's going on.

File: src/EffectManager.cpp [hunk 3]
```diff
@@ -135,3 +158,58 @@ ChainCallbackResult EffectManager::OnDraw(EffectManager *mgr)
...
+        effect->updateCallback = g_Effects[effectIdx].updateCallback;
+        effect->timer.InitializeForPopup();
+        effect->unk_17a = 0;
+        effect->unk_17b = 0;
+        count--;
+
+        if (count == 0)
+            break;
+
+        if (!this->nextIndex)
```
Comment (roblabla) [inline src/EffectManager.cpp:205]: Same change here.

**Comment by roblabla** [comment]
Looks good otherwise. Thanks for the contribution!

File: src/EffectManager.cpp [hunk 4]
```diff
@@ -135,3 +158,58 @@ ChainCallbackResult EffectManager::OnDraw(EffectManager *mgr)
...
+        effect->unk_17a = 0;
+        effect->unk_17b = 0;
+        count--;
+
+        if (count == 0)
+            break;
+
+        if (this->nextIndex == 0)
+        {
+            effect = this->effects;
```
Comment (roblabla) [inline src/EffectManager.cpp:207]: You missed this X)

Comment (Renzo904) [reply src/EffectManager.cpp:207]: Damn xd

**Pull request merged**